src/server.h

```diff
@@ -429,6 +434,26 @@ extern int configOOMScoreAdjValuesDefaults[CONFIG_OOM_COUNT];
 #define CLIENT_REPLICATION_DONE (1ULL << 51)         /* Indicate that replication has been done on the client */
 #define CLIENT_AUTHENTICATED (1ULL << 52)            /* Indicate a client has successfully authenticated */
 
+#define CLIENT_REPL_MAIN_CONN                                                                                          \
+    (1ULL << 52) /* Dual connection sync: track a connection                                                           \
+                         which is used for online replication data */
+#define CLIENT_REPL_RDB_CONN                                                                                           \
```

Comment:
on a closer look `CLIENT_REPL_RDB_CONN` is pretty much a synonym to `CLIENT_REPL_RDBONLY`. Can we fold this new state into `CLIENT_REPL_RDBONLY`? 
```suggestion
#define CLIENT_REPL_RDB_CONN                                                                                           \
```

Reply:
`CLIENT_REPL_RDBONLY` is used for other cases then RDB connection sync. If we merge it into one flag, normal clients would not be able to request a snapshot.

Reply:
Not a blocker for me for this PR but let's take a closer look separately?

Reply:
NP

Reply:
I revisited the possibility of merging the two flags `repl_rdbonly` and `repl_rdb_channl`. However, there seems to be a potential issue that needs to be considered. In the [replication.c:1176](https://github.com/valkey-io/valkey/blob/unstable/src/replication.c#L1176) section of the code, when a replica requests to join an ongoing full synchronization, it checks whether the new replica requires replication backlog. The way to determine this is by verifying the `repl_rdbonly` flag of the replica. This is where the distinction becomes crucial. When a replica has the `repl_rdb_channl` flag set, it indicates that it needs to maintain replication data for the main connection.

The same problem can also arise in the opposite scenario - if the existing replica is configured for dual-channel replication, and the new replica requires a backlog, we might reject the new replica from joining the current synchronization process, despite the primary server having the necessary backlog available.
```
void syncCommand(client *c) {
         ...        
        /* Ok a background save is in progress. Let's check if it is a good
         * one for replication, i.e. if there is another replica that is
         * registering differences since the server forked to save. */
        client *replica;
        listNode *ln;
        listIter li;

        listRewind(server.replicas, &li);
        while ((ln = listNext(&li))) {
            replica = ln->value;
            /* If the client needs a buffer of commands, we can't use
             * a replica without replication buffer. */
            if (replica->repl_state == REPLICA_STATE_WAIT_BGSAVE_END &&     // this replica is syncing
                (!(replica->flag.repl_rdbonly) || (c->flag.repl_rdbonly)))  // this replica has repl-buffer or the new replica doesn't need repl-buffer
                break;
        }
       ...
```